go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -498,14 +516,22 @@ func getPodInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config,
 	}
 
 	// Find rtos MAC (this is the pod's first hop router).
-	podInfo.RtosMAC, err = getRouterPortMacAddress(coreclient, restconfig, podInfo, ovnNamespace, types.RouterToSwitchPrefix)
+	lrpPrefix := types.RouterToSwitchPrefix
+	if podInfo.TopologyType() == types.Layer2Topology {
```

Comment:
isn't this only for primary UDN ? 

In seconday UDN there's no routers involved.